scripts/globals/treasure.lua

```diff
@@ -1465,18 +1465,19 @@ tpz.treasure.onTrade = function(player, npc, trade, chestType)
 
     -- gil
     if roll <= info.gil[1] then
-        players = player:getParty()
-        for i = 1, #players, 1 do
-            if player:getZoneID() ~= players[i]:getZoneID() then
-                table.remove(players, i)
+        local partyMembers = player:getAlliance()
+        local membersInZone = {}
+        for i = 1, #partyMembers do
+            if player:getZoneID() == partyMembers[i]:getZoneID() then
+                table.insert(membersInZone, partyMembers[i])
             end
         end
         local gilAmount = math.random(info.gil[2], info.gil[3])
-        local gil = gilAmount/#players
-        for i = 1, #players, 1 do
-            if player:getZoneID() == players[i]:getZoneID() then
-                players[i]:addGil(gil)
-                players[i]:messageSpecial(ID.text.GIL_OBTAINED, gil)
+        local gil = gilAmount/#membersInZone
+        for i = 1, #membersInZone do
+            if player:getZoneID() == membersInZone[i]:getZoneID() then
```

Comment:
You've already verified these members were in the same zone, hence the name of the table. 😉

Reply:
True! Not sure why I left that there, fixed.